libcst/codemod/commands/apply_stub_annotations.py

```diff
@@ -0,0 +1,506 @@
+        if "@" "generated" in node.value:
+            self.is_generated = True
+
+    def leave_Module(
+        self, original_node: cst.Module, updated_node: cst.Module
+    ) -> cst.Module:
+        fresh_class_definitions = [
+            definition
+            for name, definition in self.class_definitions.items()
+            if name not in self.visited_classes
+        ]
+        if self.is_generated:
+            return original_node
+        if (
+            not self.toplevel_annotations
+            and not self.imports
+            and not fresh_class_definitions
+        ):
+            return updated_node
+        toplevel_statements = []
+        # First, find the insertion point for imports
+        statements_before_imports, statements_after_imports = self._split_module(
+            original_node, updated_node
+        )
+
+        # Make sure there's at least one empty line before the first non-import
+        statements_after_imports = self._insert_empty_line(statements_after_imports)
+
+        imported = set()
+        for statement in self.import_statements:
+            names = statement.names
+            if isinstance(names, cst.ImportStar):
+                continue
+            for name in names:
+                if name.asname:
+                    name = name.asname
+                if name:
+                    imported.add(_get_name_as_string(name.name))
+
+        for _, import_statement in self.imports.items():
+            # Filter out anything that has already been imported.
+            names = import_statement.names.difference(imported)
+            names = [cst.ImportAlias(cst.Name(name)) for name in sorted(names)]
+            if not names:
+                continue
+            import_statement = cst.ImportFrom(
+                module=import_statement.module, names=names
+            )
+            # Add import statements to module body.
+            # Need to assign an Iterable, and the argument to SimpleStatementLine
+            # must be subscriptable.
+            toplevel_statements.append(cst.SimpleStatementLine([import_statement]))
+
+        for name, annotation in self.toplevel_annotations.items():
+            annotated_assign = cst.AnnAssign(
+                cst.Name(name),
+                # pyre-fixme[16]: `CSTNode` has no attribute `annotation`.
+                cst.Annotation(annotation.annotation),
+                None,
+            )
+            toplevel_statements.append(cst.SimpleStatementLine([annotated_assign]))
+
+        toplevel_statements.extend(fresh_class_definitions)
+
+        return updated_node.with_changes(
+            body=[
+                *statements_before_imports,
+                *toplevel_statements,
+                *statements_after_imports,
+            ]
+        )
+
+
+def apply_stub_annotations(stubs: cst.Module, source: cst.Module) -> cst.Module:
+    import_visitor = ImportCollector()
+    source.visit(import_visitor)
```

Comment:
We have a helper `GatherImportsVisitor` for collecting imports. Can we reuse it to be simpler? https://libcst.readthedocs.io/en/latest/codemods.html#libcst.codemod.visitors.GatherImportsVisitor